component/service/src/main/java/org/exoplatform/social/rest/impl/user/UserRestResourcesV1.java

```diff
@@ -296,9 +296,17 @@ public Response getUsers(@Context UriInfo uriInfo,
       filter.setSkills(q == null || q.isEmpty() ? "" : q);
       filter.setEnabled(!isDisabled);
       filter.setUserType(userType);
+      filter.setSearchEmail(true);
       if (isDisabled && q != null && !q.isEmpty()) {
+        ListAccess<User> usersListAccess;
         User[] users;
-        ListAccess<User> usersListAccess = userSearchService.searchUsers(q, UserStatus.DISABLED);
+        if(q.contains("@")) {
```

Comment:
as userSearchService calls orgSe'rvice.getUserHandler.findUsersByQuery, it is possible to remove this if ?
I dont understand why we make a different treatment when query contains @ or not, can you explain ?

Reply:
to be treated in the new US as explained by @ahmadmido